drivers/adodb-db2.inc.php

```diff
@@ -198,12 +205,28 @@ private function unpackParameters($argDSN, $argUsername, $argPassword, $argDatab
 	{
 
 
-		$connectionParameters = array('dsn'=>'',
-									  'uid'=>'',
-									  'pwd'=>'',
-									  'database'=>'',
-									  'catalogue'=>true
-									  );
+		$connectionParameters = array(
+			'dsn'=>'',
+			'uid'=>'',
+			'pwd'=>'',
+			'database'=>'',
+			'catalogue'=>true
+		);
+
+		/*
+		* Shortcut for *LOCAL
+		*/
+		if (strcmp($argDSN,'*LOCAL') == 0)
+		{
+			$connectionParameters = array(
+				'dsn'=>'*LOCAL',
+				'uid'=>'',
+				'pwd'=>'',
+				'database'=>'',
+				'catalogue'=>true
+			);
+			return $connectionParameters;
+		}
```

Comment:
Suggest moving the _if *LOCAL_ block to the top of the function, there is no point in setting $connectionParameters and immediately overwriting its value and return.